reliquery/storage.py

```diff
@@ -505,6 +511,463 @@ def get_all_relic_data(self) -> List[Dict]:
             return relic_data
 
 
+class GoogleDriveStorage(Storage):
+    def __init__(
+        self,
+        prefix: str,
+        name: str,
+        token_file: str,
+        SCOPES: list,
+        shared_folder_id: str,
+    ):
+        self.prefix = prefix
+        self.name = name
+        self.token_file = token_file
+        self.SCOPES = SCOPES
+        self.shared_folder_id = shared_folder_id
+
+        creds = None
+        if os.path.exists(token_file):
+            creds = service_account.Credentials.from_service_account_file(
+                token_file, scopes=SCOPES
+            )
+        try:
+            self.service = build("drive", "v3", credentials=creds)
+        except HttpError as error:
+            print(f"An error occurred: {error}")
+
+        # Check for prefix folder
+        results = (
+            self.service.files()
+            .list(
+                q="parents in '{}'".format(self.shared_folder_id),
+                pageSize=100,
+                fields="nextPageToken, files(id, name)",
+            )
+            .execute()
+        )
+        items = results.get("files", [])
+
+        self.root_id = ""
+        prefix_found = False
+        for item in items:
+            if item["name"] == prefix:
+                self.root_id = item["id"]
+                prefix_found = True
+                break
+
+        if prefix_found is False:
```

Comment:
Could we use the _create_folder() defined below here to create the prefix dir? Looks like there is almost the same code in both bodies.